src/Loader/AnnotationLoader.php

```diff
@@ -44,6 +45,14 @@ public function getMetadataFor(string $className): ClassMetadataInfo
             $metadata->addPropertyMetadata($propertyMetadata);
         }
 
+        foreach ($metadata->reflection->getMethods() as $method) {
```

Comment:
You need to check the visibility and if they don't are abstract.

Reply:
In case of visibility I don't be sure. We can use `\ReflectionMethod::IS_PUBLIC` filter but maybe in some cases its better to restrict visibility to not accidentally call the method from the outside.

I dont see any needs to check for a abstract method. A class with abstract methods must be abstract, so it would not be possible to instantiate it. If a class implements a abstract class with abstract methods the concrete class must implement thoes methods.